config/_default/menus/main.en.yaml

```diff
@@ -3994,171 +3994,246 @@ menu:
       identifier: custom_instrumentation_go
-      weight: 2024
+      weight: 2034
     - name: Node.js
       url: tracing/trace_collection/custom_instrumentation/nodejs/otel
       parent: custom_instrumentation
       identifier: custom_instrumentation_nodejs
-      weight: 2025
+      weight: 2035
     - name: PHP
       url: tracing/trace_collection/custom_instrumentation/php/otel
       parent: custom_instrumentation
       identifier: custom_instrumentation_php
-      weight: 2026
+      weight: 2036
     - name: C++
       url: tracing/trace_collection/custom_instrumentation/cpp/dd-api
       parent: custom_instrumentation
       identifier: custom_instrumentation_cpp
-      weight: 2027
+      weight: 2037
     - name: .NET
       url: tracing/trace_collection/custom_instrumentation/dotnet/otel
       parent: custom_instrumentation
       identifier: custom_instrumentation_dotnet
-      weight: 2028
+      weight: 2038
     - name: Android
       url: tracing/trace_collection/custom_instrumentation/android/otel
       parent: custom_instrumentation
       identifier: custom_instrumentation_android
-      weight: 2029
+      weight: 2039
     - name: iOS
       url: tracing/trace_collection/custom_instrumentation/ios/otel
       parent: custom_instrumentation
       identifier: custom_instrumentation_ios
-      weight: 2030
+      weight: 2040
     - name: Rust
       url: tracing/trace_collection/custom_instrumentation/rust
       parent: custom_instrumentation
       identifier: custom_instrumentation_rust
-      weight: 2031
+      weight: 2041
     - name: Swift
       url: tracing/trace_collection/custom_instrumentation/swift
       parent: custom_instrumentation
       identifier: custom_instrumentation_swift
-      weight: 2032
+      weight: 2042
     - name: Elixir
       url: tracing/trace_collection/custom_instrumentation/elixir
       parent: custom_instrumentation
       identifier: custom_instrumentation_elixir
-      weight: 2033
+      weight: 2043
+    - name: Dynamic Instrumentation
+      url: tracing/trace_collection/dynamic_instrumentation/
+      identifier: dyninst
+      parent: tracing_trace_collection
+      weight: 204
+    - name: Autocomplete and Search
+      url: tracing/trace_collection/dynamic_instrumentation/symdb
+      identifier: dyninst_symdb
+      parent: dyninst
+      weight: 2041
+    - name: Expression Language
+      url: tracing/trace_collection/dynamic_instrumentation/expression-language
+      identifier: dyninst_explang
+      parent: dyninst
+      weight: 2042
+    - name: Sensitive Data Scrubbing
+      url: tracing/trace_collection/dynamic_instrumentation/sensitive-data-scrubbing
+      identifier: dyninst_sensitive_data_scrubbing
+      parent: dyninst
+      weight: 2043
     - name: Library Compatibility
       url: tracing/trace_collection/compatibility/
       parent: tracing_trace_collection
       identifier: tracing_compatibility
-      weight: 203
+      weight: 206
```

Comment:
I noticed that other parent topics at this level (Single Step Instrumentation, for example) have nested topics in the left nav, but this one doesn't. I don't think we necessarily need to, especially if this was a conscious decision, but thought I would bring up the inconsistency just in case it was something you wanted to make consistent. If so, it could easily go into a separate PR down the road too!